### PR TITLE
Add verification hooks for lowering configuration.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**
This PR allows for checking if the `#iree_codegen.translation.info`
and `#iree_codegen.lowering.config` on an operation are valid for a
given pipeline. Apart from verifying existing configurations, it also
allows a search to prune out illegal configurations.

### CODE REVIEW

File: iree/compiler/Codegen/LLVMCPU/Passes.cpp [hunk 1]
```diff
@@ -48,6 +48,87 @@ void addCPUVectorizationPassPipeline(OpPassManager &passManager,
...
+               << val.index() << " and workload_per_wg value " << val.value();
+      }
+    }
+  }
+
+  // Verify that native vector size is either empty, or if set is same as the
+  // last level of tiling
+  SmallVector<int64_t> nativeVectorSize =
+      loweringConfig.getNativeVectorSizeVals();
+  if (!nativeVectorSize.empty()) {
```
Comment (hanhanW) [inline iree/compiler/Codegen/LLVMCPU/Passes.cpp:123]: I think this is already checked by `LoweringConfigAttr::verify` itself?

https://github.com/google/iree/blob/dacf2b19798d2251409d78b0faf88d07756faa26/iree/compiler/Codegen/Dialect/LoweringConfig.cpp#L158-L175

**Review by hanhanW — APPROVED** [review]
overall looks good, just few nits

File: iree/compiler/Codegen/LLVMCPU/Passes.cpp (hunk 1 above)
Comment (MaheshRavishankar) [reply iree/compiler/Codegen/LLVMCPU/Passes.cpp:124]: Thats the `tileSizes`. not the `nativeVectorSize`.

File: iree/compiler/Codegen/Passes.h [hunk 2]
```diff
@@ -19,9 +20,15 @@
 namespace mlir {
 namespace iree_compiler {
 
-// Registers all conversion passes in this directory.
+/// Registers all conversion passes in this directory.
 void registerCodegenPasses();
 
+/// Verify that the configuration used for compilation is valid.
+LogicalResult verifyLoweringConfiguration(
```
Comment (antiagainst) [inline iree/compiler/Codegen/Passes.h:27]: If I read it correctly, this isn't really used and can be deleted?

**Review by antiagainst — APPROVED** [review]

File: iree/compiler/Codegen/LLVMCPU/Passes.cpp (hunk 1 above)
Comment (hanhanW) [reply iree/compiler/Codegen/LLVMCPU/Passes.cpp:124]: It truncates what I'd like to show. If you scroll it down, you'll see this. But now I found that you're correct. You are checking if it is empty. This is just check that it should exist.

https://github.com/google/iree/blob/dacf2b19798d2251409d78b0faf88d07756faa26/iree/compiler/Codegen/Dialect/LoweringConfig.cpp#L172-L175

File: iree/compiler/Codegen/Passes.h (hunk 2 above)
Comment (MaheshRavishankar) [reply iree/compiler/Codegen/Passes.h:27]: This is the external hook for auto-tuners to verify the configuration.

**Pull request merged**